KubeArmor/eventAuditor/sharedMapManager.go

```diff
@@ -1,8 +1,146 @@
 package eventAuditor
 
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	lbpf "github.com/aquasecurity/tracee/libbpfgo"
+)
+
 // =========================== //
 // == Shared Map Management == //
 // =========================== //
 
-// handle process-spec table
+// eventAuditorEBPFModule Structure
+type eventAuditorEBPFModule struct {
+	eaMod *lbpf.Module
+	eaMap *lbpf.BPFMap
+}
+
+var sharedMods map[string]*eventAuditorEBPFModule
```

Comment:
@nam-jaehyun 

BPFMap has a private pointer for the Module that contains it, but libbpfgo doesn't provide it for the consumer of the BPFMap. The solution that I found was to maintain the Map and its respective Module on our eventAuditorEBPFModule.

Why am I doing this? Well I think it's better to have an unique module for every single pinned map. This would make maintenance easier.

Reply:
This workaround won't be necessary anymore. The https://github.com/aquasecurity/libbpfgo/pull/41 solved this question.